pkg/repository/git.go

```diff
@@ -74,6 +76,27 @@ func (r gitRepoController) calculateIsClean(ctx context.Context) (bool, error) {
 	return status.IsClean(), nil
 }
 
+// GetBranch returns the branch from a .git directory
+func (r gitRepoController) getBranch() (string, error) {
+	repo, err := r.repoGetter.get(r.path)
+	if err != nil {
+		return "", fmt.Errorf("failed to analyze git repo: %w", err)
+	}
+
+	head, err := repo.Head()
+	if err != nil {
+		return "", fmt.Errorf("failed to infer the git repo's current branch: %w", err)
+	}
+
+	branch := head.Name()
```

Comment:
```suggestion
	ref := head.Name()
```
I think this is not a branch